custom/icds/forms.py

```diff
@@ -91,6 +91,11 @@ def __init__(self, request, domain, email, *args, **kwargs):
             )
         )
 
+    def clean_link_id(self):
+        if not self.cleaned_data.get('link_id'):
+            self.add_error('link_id', _("Please select link"))
```

Comment:
Confused by this - if the link is optional now, why are you showing an error when it isn't provided?

Reply:
This is because there is a single form for both search and new record. We need this in case of POST requests.

Reply:
aha, got it